paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -243,4 +246,19 @@ public void rollbackTo(long snapshotId) {
             throw new UncheckedIOException(e);
         }
     }
+
+    @Override
+    public void createTag(String tagName, long fromSnapshotId) {
+        checkArgument(!StringUtils.isNumeric(tagName), "Tag name cannot be pure numeric string.");
```

Comment:
inline `isNumeric` here.
And add test for empty "" tag string.
Move this check to `TagManager`.